### PR TITLE
#764 Fix VASSAL deprecated code

### PR DESCRIPTION
**Pull request by mtl1971**

### CODE REVIEW

File: src/VASL/build/module/map/boardPicker/Underlay.java [hunk 1]
```diff
@@ -59,15 +61,17 @@ public void readData() {
...
       underlayImage = ImageIO.read(new MemoryCacheImageInputStream(in));
     }
     catch (IOException ex) {
     }
 
     if (underlayImage == null) {
-      try {
-        underlayImage = archive.getImage(imageName);
+      // ... or try to load it from the one passed in from the board
+      try (InputStream archiveIn = archive.getInputStream(imageName)) {
```
Comment (uckelman) [inline src/VASL/build/module/map/boardPicker/Underlay.java:73]: I'm not convinced this is correct now, nor am I convinced this was correct before. The question to be answered first is: Where can underlay images be?

Comment (mtl1971) [reply src/VASL/build/module/map/boardPicker/Underlay.java:73]: You're right. Just looked through all the boards/src files for underlays - in every case, they only ever reference files in dist/boardData - so the second attempt appears to be defunct.

Comment (uckelman) [reply src/VASL/build/module/map/boardPicker/Underlay.java:73]: Well, I'm not sure. Someone who is familiar with where the underlays are needs to comment.

**Comment by derimmer** [comment]
Underlays are referenced in the BoardMetaData.xml files found in each bdXX file. The actual images are found in the dist/BoardData directory of the VASL module. They are used by the Underlay.java class. They are also referenced in the OverlaySSR file found in bdXX files. This is the old way of referencing them and has been replaced by the use of the BoardMetaData.xml file. I am just about to start removing OverlaySSR files from bdXX files. See issue #836.

Not all boards have BoardMetaData.xml files. Those that don't are usually older or specialized boards. Most of the geomorphic boards have a BoardMetaData.xml file. But even if they do, they may not use Underlays. 

Underlays support terrain transformations set by the TerrainSSR button in the BoardPicker wizard.

Overlay files can be found in two places. For overlays that are added to a board using the Overlay button in the BoardPicker wizard, these are found in the boards/overlays directory on the users device, which is set by the user. The overlay files found in this directory are also .zip files without the suffix. They contain images which are added to the board. They are used by the Overlay.java class. Board specific overlays can be found in the bdXX file for the board to which they apply. Like Underlays they are referenced in the BoardMetaData.xml file. They are used by the SSROverlay.java file. Board specific overlays are also referenced in the OverlaySSR files found in bdXX files. As above with underlays, this is now unnecessary and I will be deleting the files.  

I am not aware of the rationale for underlays versus SSRoverlays but I suspect it has to do with the order in which multiple terrain transformations are applied to boards. See issue #70. Perhaps @davidsullivan317 can help?

Sorry for the length of this but as you can see there are several different places in which images are stored. 

For completeness sake: there is also an overlay extension available. This uses a completely different approach to adding overlays via drag-and-drop. Essentially it creates overlays as counters. It does not use any of the deprecated code that we are trying to address.

**Comment by mtl1971** [comment]
Thanks for that, Doug. So just considering Underlays, the actual underlay images are only ever to be found in dist/boardData?

**Comment by derimmer** [comment]
> Thanks for that, Doug. So just considering Underlays, the actual underlay images are only ever to be found in dist/boardData?

As far as I know, yes.

**Comment by mtl1971** [comment]
Ok, I've added some logging to capture failed attempts to load Underlays, just in case. I've created a several maps requiring underlays now, and they are all constructed successfully each time with the underlay images loaded correctly.

**Pull request merged**